data_discovery_ai/utils/es_connector.py

```diff
@@ -43,32 +43,72 @@ def connect_es(config: configparser.ConfigParser) -> Elasticsearch:
 
 def search_es(client: Elasticsearch):
     index = "es-indexer-staging"
-    start_index = 0
     dataframes = []
     batch_size = 100
-    resp = client.search(index=index, body={"query": {"match_all": {}}})
-    total_number = resp["hits"]["total"]["value"]
 
+    # get document count
+    count_resp = client.count(index=index)
+    total_number = count_resp["count"]
     rounds = (total_number + batch_size - 1) // batch_size
 
-    # TODO: refactor the search query method.
-    for round in tqdm(range(rounds), desc="searching elasticsearch"):
-        search_query = {
-            "size": batch_size,
-            "from": start_index,
+    # get the first pit
+    first_pit_resp = client.open_point_in_time(
+        index=index,
+        keep_alive="1m",
+    )
+    first_pit = first_pit_resp["id"]
+
+    # the first search
+    first_query_body = {
+        "size": 100,
+        "query": {"match_all": {}},
+        "pit": {"id": first_pit, "keep_alive": "1m"},
+        "sort": [{"id.keyword": "asc"}],
+    }
+    first_query_resp = client.search(body=first_query_body)
+
+    data = first_query_resp["hits"]["hits"]
+    # set search after value
+    current_last_result = data[-1]["sort"]
+
+    # save the first search result
+    df = pd.json_normalize(data)
+    dataframes.append(df)
+
+    # set current pit as the first one
+    current_pit = first_query_resp["pit_id"]
+
+    # conduct further search
+    for round in tqdm(range(1, rounds), desc="searching elasticsearch"):
+        qurry_body = {
+            "size": 100,
             "query": {"match_all": {}},
+            "pit": {"id": current_pit, "keep_alive": "1m"},
+            "sort": [{"id.keyword": "asc"}],
+            "search_after": current_last_result,
+            "track_total_hits": False,
         }
+        query_resp = client.search(body=qurry_body)
 
-        resp = client.search(index=index, body=search_query)
+        data = query_resp["hits"]["hits"]
+        # set search after value
+        current_last_result = data[-1]["sort"]
 
-        data = resp["hits"]["hits"]
+        # save the first search result
         df = pd.json_normalize(data)
         dataframes.append(df)
 
-        start_index += 1
+        # set current pit as the first one
+        current_pit = query_resp["pit_id"]
+
         round += 1
-        time.sleep(1)
+        time.sleep(10)
```

Comment:
why 10?

Reply:
and is hard-coded, maybe make it a parameter of `search_es()` method, together with the batch size `100` I commented above, should become

```python
def search_es(client: Elasticsearch, index: str = "es-indexer-staging", batch_size: int = 100, sleep_time: int = 10):
...
```